src/xb-value-bindings.c

```diff
@@ -17,30 +17,23 @@
 #include <glib.h>
 
 #include "xb-opcode-private.h"
+#include "xb-silo-private.h"
 #include "xb-value-bindings-private.h"
 
 typedef struct {
-	enum {
-		KIND_NONE,
-		KIND_TEXT,
-		KIND_INTEGER,
-	} kind;
-	union {
-		gchar *text;
-		guint32 integer;
-	};
```

Comment:
Why get rid of the union? Now the struct is a bit bigger, which is important because it’s supposed to be used on the stack. The bigger it is, the less likely it’ll fit in L1, hence it might be slower.

Reply:
OK, makes sense